lib/Extractor/ExprBuilder.cpp

```diff
@@ -376,6 +376,23 @@ Inst *ExprBuilder::getDemandedBitsCondition(Inst *I) {
                                    LIC->getInst(Inst::Eq, 1, {Res, Zero})});
     Result = LIC->getInst(Inst::And, 1, {Result, SignBits});
   }
+  // TODO: We might want to look into handling empty set later
+  if (!I->Range.isEmptySet() && !I->Range.isFullSet()) {
+    Inst *Lower = LIC->getConst(I->Range.getLower());
+    Inst *Upper = LIC->getConst(I->Range.getUpper());
+ 
+    // not wrapped set: x >= Lower && x <= Upper
+    // wrapped set: x >= Lower || x <= Upper
+    if (!I->Range.isWrappedSet()) {
+      Result = LIC->getInst(Inst::And, 1, {Result, LIC->getInst(Inst::And, 1,
+                            {LIC->getInst(Inst::Ule, 1, {Lower, I}),
+                             LIC->getInst(Inst::Ult, 1, {I, Upper})})});
+    } else if (I->Range.isWrappedSet()) {
```

Comment:
why are you checking !isWrappedSet() and then also isWrappedSet()??

Reply:
I did that earlier because I thought of handling isSignedWrappedSet() separately, but it's not handled here and is just an else case now.